tests/testthat/helper.R

```diff
@@ -43,75 +43,54 @@ map_numbers_to_new_range <- function(numbers, lower, upper) {
 
 }
 
-#' A helper function to get all files in the ~/R/ folder
+#' A helper function that returns the full paths of the package files as a vector
 #' @keywords internal
 
-get_R_files <- function() {
+.get_visR_files <- function(functions = FALSE,
+                            tests = FALSE,
+                            documentation = FALSE,
+                            vignettes = FALSE,
+                            remove_watchdog = TRUE) {
 
-  R_files <- base::list.files(path = base::paste0(base::getwd(), "/../../R"),
-                                pattern = "*.R",
-                                full.names = TRUE)
-
-  return(unlist(R_files))
-
-}
-
-#' A helper function to get all files in the ~/tests/testthat/ folder
-#' @keywords internal
-
-get_test_files <- function() {
-
-  test_files <- base::list.files(path = base::getwd(),
-                                 pattern = "*.R",
-                                 full.names = TRUE)
-
-  return(unlist(test_files))
-
-}
-
-#' A helper function to get all files in the ~/man/ folder
-#' @keywords internal
+  files <- list()
+  wd <- getwd()
 
-get_man_files <- function() {
+  if (functions) {
 
-  man_files <- base::list.files(path = base::paste0(base::getwd(), "/../../man"),
-                                pattern = "*.Rd",
-                                full.names = TRUE)
+    R_files <- list.files(path = paste0(wd, "/../../R"), 
+                          pattern = "*.R", 
+                          full.names = TRUE)
+    files <- c(files, unlist(R_files))
 
-  return(unlist(man_files))
+  }
 
-}
+  if (tests) {
```

Comment:
Dont you need the files from the test folder?

Reply:
Yes, because I think that if we already are enforcing the CRAN mandated conventions in some places, we should just enforce them in all places